### ISSUE TITLE
1.2.6.17 Rebuy Preventing buys and new volume setting preventing buys as well...

### ISSUE THREAD

**Issue by TheBigShort111** [taniman/profit-trailer#1137]
It i my understanding that in order to upgrade from one version to another, all that is required is to replace the .jar file - However in this last update, it seems that simply replacing the .jar file has had unexpected results... It seems some new settings were added that I am not seeing in my current .16 DCA folder - Also, it seems this issue could have some sort of compatibility issue with feeder if that is the case.

So, I'm wondering if I need to replace all files for this latest update to avoid the issues, or is this related to feeder taking over the Pairs and DCA files?

2018-02-17 00:14:58 INFO ProfitTrailerService - Cannot buy BTCUSDT. rebuy timeout active!
2018-02-17 00:15:03 INFO ProfitTrailerService - Cannot buy NEOUSDT. rebuy timeout active!
2018-02-17 00:15:09 INFO NormalStrategyRunner - There is not enough volume to buy NEOUSDT
2018-02-17 00:15:56 INFO NormalStrategyRunner - There is not enough volume to buy BTCUSDT
2018-02-17 00:16:07 INFO ProfitTrailerService - Cannot buy NEOUSDT. rebuy timeout active!
2018-02-17 00:16:08 INFO ProfitTrailerService - Cannot buy BTCUSDT. rebuy timeout active!
2018-02-17 00:16:13 INFO NormalStrategyRunner - There is not enough volume to buy ETHUSDT
2018-02-17 00:16:19 INFO ProfitTrailerService - Cannot buy ETHUSDT. rebuy timeout active!
2018-02-17 00:16:19 INFO ProfitTrailerService - Cache Heartbeat
2018-02-17 00:16:31 INFO NormalStrategyRunner - There is not enough volume to buy NEOUSDT
2018-02-17 00:16:38 INFO ProfitTrailerService - Cannot buy BCCUSDT. rebuy timeout active!
2018-02-17 00:17:08 INFO ProfitTrailerService - Cannot buy NEOUSDT. rebuy timeout active!
2018-02-17 00:17:13 INFO ProfitTrailerService - Cannot buy LTCUSDT. rebuy timeout active!
2018-02-17 00:17:19 INFO NormalStrategyRunner - There is not enough volume to buy LTCUSDT
2018-02-17 00:17:30 INFO ProfitTrailerService - Normal Heartbeat
2018-02-17 00:17:37 INFO ProfitTrailerService - DCA Heartbeat
2018-02-17 00:17:43 INFO ProfitTrailerService - Cannot buy BCCUSDT. rebuy timeout active!
2018-02-17 00:17:49 INFO NormalStrategyRunner - There is not enough volume to buy NEOUSDT
2018-02-17 00:17:49 INFO NormalStrategyRunner - There is not enough volume to buy BCCUSDT
2018-02-17 00:18:13 INFO ProfitTrailerService - Cannot buy NEOUSDT. rebuy timeout active!
2018-02-17 00:18:48 INFO ProfitTrailerService - Cannot buy BCCUSDT. rebuy timeout active!
2018-02-17 00:19:00 INFO NormalStrategyRunner - There is not enough volume to buy NEOUSDT
2018-02-17 00:19:17 INFO ProfitTrailerService - Cannot buy NEOUSDT. rebuy timeout active!

**Comment by SgtBatten** [comment]
Adding and removing of parameters is always included in the change log and is manually required to be edited by the user. Otherwise you lose you let setting by replacing the whole file. 

We don't include all available setting in the default download.

We also keep this list updated of parameter changes. https://wiki.profittrailer.io/doku.php/changes

**Comment by emx0r** [comment]
@SgtBatten can it be documented better way next time?
I mean I did not have these new options set after the upgrade, yet I got the message that rebuy is on hold:

2018-02-17 14:07:09 INFO ProfitTrailerService - Cannot buy BTC-ZCL. rebuy timeout active!

Looking at wiki and upgrade instructions, there is nothing mentioned what is the default if options are missing. Also the description could be better it's logical to assume the value is in minutes but where is it indicated on wiki?

Also when adding new options would not be better to keep default to work like in previous version, so people don't get hit by something they do not fully understand and don't expect something like that can happen.

**Comment by SgtBatten** [comment]
Step 4 of the upgrade instructions quite explicitly says to check the release notes and the changes page for new parameters, parameters that have been renamed etc.

The pairs and DCA wiki entries also describe every parameter that exists in the latest version. Including rebuy timeout stating that it does indeed refer to minutes. 

The only thing we didn't include is what the default value is when it's not written into your files. I admit I am waiting on clarification of this before I edit the wiki

**Comment by acefed** [comment]
A default value 0 would have prevented all these issues people are reporting :)

**Comment by SgtBatten** [comment]
It was not intentional. And default is already 0 in version 18 now in testing

**Comment by acefed** [comment]
I know, these things happen sometimes. I am glad with all the work and new options you guys work on. Keep up the good work!
